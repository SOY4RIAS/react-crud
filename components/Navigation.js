import React from 'react'
import Link from 'next/link'
import { IoIosAdd } from 'react-icons/io'
import PWAInstallButton from './PWAInstallButton'

const Navigation = () => (
  <nav className="fixed w-full bottom-0 z-50 sm:relative">
    <ul className="flex justify-between items-center bg-gray-200 px-4 py-2 text-lg uppercase">
      <li>
        <Link href="/">
          <a className="block">My Notes</a>
        </Link>
      </li>
      <li>
        <PWAInstallButton />
      </li>
      <li>
        <Link href="/add">
          <a
            className="block text-5xl text-green-600"
            title="Add Note"
            data-cy="nav-add-note"
          >
            <IoIosAdd />
          </a>
        </Link>
      </li>
    </ul>
  </nav>
)

export default Navigation
