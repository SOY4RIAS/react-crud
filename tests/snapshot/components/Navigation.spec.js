import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Navigation from '~/components/Navigation'

describe('Navigation', () => {
  it('should render the navigation', () => {
    const { container } = render(<Navigation />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
