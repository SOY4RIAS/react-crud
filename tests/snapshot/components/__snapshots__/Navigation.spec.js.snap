// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render the navigation 1`] = `
<nav
  class="fixed w-full bottom-0 z-50 sm:relative"
>
  <ul
    class="flex justify-between items-center bg-gray-200 px-4 py-2 text-lg uppercase"
  >
    <li>
      <a
        class="block"
        href="/"
      >
        My Notes
      </a>
    </li>
    <li>
      <a
        class="block text-5xl text-green-600"
        data-cy="nav-add-note"
        href="/add"
        title="Add Note"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M368.5 240H272v-96.5c0-8.8-7.2-16-16-16s-16 7.2-16 16V240h-96.5c-8.8 0-16 7.2-16 16 0 4.4 1.8 8.4 4.7 11.3 2.9 2.9 6.9 4.7 11.3 4.7H240v96.5c0 4.4 1.8 8.4 4.7 11.3 2.9 2.9 6.9 4.7 11.3 4.7 8.8 0 16-7.2 16-16V272h96.5c8.8 0 16-7.2 16-16s-7.2-16-16-16z"
          />
        </svg>
      </a>
    </li>
  </ul>
</nav>
`;
